src/transformers/models/vocos/modeling_vocos.py

```diff
@@ -0,0 +1,413 @@
+            window_sqrt,
+            output_size=(1, output_length),
+            kernel_size=(1, win_length),
+            stride=(1, hop_length),
+        ).squeeze()[pad:-pad]
+
+        if torch.any(norm <= 1e-11):
+            raise ValueError(
+                "Normalization tensor `norm` contains values ≤ 1e-11, it would cause division by zero. check the n_fft, hop_length and padding parameters."
+            )
+
+        audio = audio / norm
+
+    return audio
+
+
+class VocosAdaptiveLayerNorm(nn.Module):
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.eps = config.layer_norm_eps
+        self.hidden_dim = config.hidden_dim
+        self.weight = nn.Parameter(torch.ones(config.adanorm_num_embeddings, config.hidden_dim))
+        self.bias = nn.Parameter(torch.zeros(config.adanorm_num_embeddings, config.hidden_dim))
+
+    def forward(self, hidden_states: torch.Tensor, cond_embedding_id: torch.LongTensor = None):
+        if cond_embedding_id is None:
+            # the index used to select the target bandwidth is used to index into Adaptive Normalization lookup table (adanorm_num_embeddings, hidden_dim)
+            raise ValueError(
+                "When using adaptive LayerNorm `use_adaptive_norm=True`, you must pass conditional id via `bandwidth_id`."
+            )
+
+        hidden_states = F.layer_norm(hidden_states, (self.hidden_dim,), weight=None, bias=None, eps=self.eps)
+        return hidden_states * self.weight[cond_embedding_id].unsqueeze(0) + self.bias[cond_embedding_id].unsqueeze(0)
+
+
+class VocosConvNeXtBlock(nn.Module):
+    """ConvNeXt block adapted for 1D convolutions in the Vocos architecture."""
+
+    def __init__(self, config: VocosConfig):
+        super().__init__()
+        self.dwconv = nn.Conv1d(
+            config.hidden_dim,
+            config.hidden_dim,
+            kernel_size=config.kernel_size,
+            padding=config.padding,
+            groups=config.hidden_dim,
+        )
+        if config.use_adaptive_norm:
+            self.norm = VocosAdaptiveLayerNorm(config)
+        else:
+            self.norm = nn.LayerNorm(config.hidden_dim, eps=config.layer_norm_eps)
+        self.pwconv1 = nn.Linear(config.hidden_dim, config.intermediate_dim)
+        self.act = nn.GELU()
+        self.pwconv2 = nn.Linear(config.intermediate_dim, config.hidden_dim)
+        if config.layer_scale_init_value > 0:
+            self.layer_scale_parameter = nn.Parameter(
+                config.layer_scale_init_value * torch.ones(config.hidden_dim), requires_grad=True
+            )
+        else:
+            self.layer_scale_parameter = None
+
+    def forward(self, hidden_states: torch.Tensor, bandwidth_id: Optional[torch.LongTensor] = None) -> torch.Tensor:
+        residual = hidden_states
+        hidden_states = self.dwconv(hidden_states)
+        hidden_states = hidden_states.transpose(1, 2)
+        if isinstance(self.norm, VocosAdaptiveLayerNorm):
```

Comment:
why not store `self.use_adaptive_norm = config.use_adaptive_norm` in `__init__` and can then check this variable?